contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y

```diff
@@ -1957,8 +1957,14 @@ AexprConst:
 				{
 					/* This is to support N'str' in various locations */
 					TypeName *t = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("nvarchar")));
+					/* Include a typmod based on the length of the literal */
+					int32 typmod = strlen($2);
+					if (typmod == 0)
+						typmod = 2; /* typmod can't be 0 */
+					else if (typmod > 4000)
+						typmod = TSQLMaxTypmod;
```

Comment:
Good point, I initially thought it should be length + VARHDRSZ as well. However, when casting to NVARCHAR, anychar_typmodin as the input function for the typmod. This is useful for cases where we explicitly set the typmod for VARCHAR and NVARCHAR, i.e. NVARCHAR(15). The input typmod would be 15, and anychar_typmodin would handle ensuring that the given value is valid, and is adjusted with VARHDRSZ. As such, we shouldn't add VARHDRSZ here since it will be added later.